pandas/core/internals/managers.py

```diff
@@ -1678,18 +1678,23 @@ def create_block_manager_from_blocks(blocks, axes):
     except ValueError as e:
         blocks = [getattr(b, "values", b) for b in blocks]
         tot_items = sum(b.shape[0] for b in blocks)
-        construction_error(tot_items, blocks[0].shape[1:], axes, e)
+        raise construction_error(tot_items, blocks[0].shape[1:], axes, e)
 
 
-def create_block_manager_from_arrays(arrays, names, axes):
+def create_block_manager_from_arrays(
+    arrays, names: Index, axes: List[Index]
+) -> BlockManager:
+    assert isinstance(names, Index)
```

Comment:
In theory mypy will flag violations here already for us, so superfluous to have these here

Reply:
@simonjayhawkins is this correct?  in a previous pass I found there were annotation-defying types being passed in this function that mypy had not caught

Reply:
yes, @WillAyd is correct, but in practice we probably need the run-time asserts to be sure. probably better to fail than potentially produce incorrect results.

> in a previous pass I found there were annotation-defying types being passed in this function that mypy had not caught

if a variable is Any or the callsite is not being checked, then we can pass the wrong type.